modules/layers/src/geojson-layer/geojson-layer.ts

```diff
@@ -50,10 +51,19 @@ import {createLayerPropsFromFeatures, createLayerPropsFromBinary} from './geojso
 
 /** All properties supported by GeoJsonLayer */
 export type GeoJsonLayerProps<DataT extends Feature = Feature> = _GeoJsonLayerProps<DataT> &
-  CompositeLayerProps<DataT>;
+  Omit<CompositeLayerProps<DataT>, 'data'>;
 
 /** Properties added by GeoJsonLayer */
 export type _GeoJsonLayerProps<DataT extends Feature = Feature> = {
+  /**
+   * The GeoJSONLayer accepts any of the following formats passed to the data prop:
+   *   * A valid GeoJSON `FeatureCollection`, `Feature`, `Geometry` or `GeometryCollection` object.
+   *   * An array of GeoJSON Feature objects.
+   *   * An URL or Promise that resolves to the above formats.
+   *   * loaders.gl's flat GeoJSON format.
+   */
+  data: LayerDataProp<Feature> | Feature | Geometry | FeatureCollection | GeometryCollection;
```

Comment:
To me it looks like the `LayerDataProp` needs two type arguments, one for the full data, and one for the data batch type:

```suggestion
  data: LayerDataProp<Feature | Geometry | FeatureCollection | GeometryCollection, Feature>;
```

Side note:t the deck.gl design with an async iterable over single features/rows is different from loaders.gl which serves async iterables over batches of rows, something like: 

```suggestion
  data: LayerDataProp<Feature | Geometry | FeatureCollection | GeometryCollection, Feature[]>;
```

Not suggesting a change here, but ideally we'll find a way to make the two systems interact transparently.